ballerina/utils.bal

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Describes Conversion Error.
+public type ConversionError distinct error<record {|
+
+    # The reason for the conversion error
+    string reason;
```

Comment:
Is this really needed? Doesn't message cover this?

Reply:
Removed the `ConversionError` and used `public type Error distinct error;` to represent error